draft-ietf-privacypass-architecture.md

```diff
@@ -683,76 +717,47 @@ must balance this against the reduction in Client privacy. In general,
-completely as it would if only 4 Issuers were permitted across the whole
-system. For example, these 4 Issuers could be different for each Client.
-Therefore, the selection of Issuers for which a Client possesses tokens is
-still revealing. This trade-off is important in deciding the effective
-anonymity of each Client in the system.
-
-Clients SHOULD bound the number of Issuers they are willing to request tokens
-from at any given time. The exact bound depends on the deployment model and
-number of Clients, i.e., having a very large Client base could potentially
-allow for larger values. Issuer replacements should only occur with the same
-frequency as config rotations as they can lead to similar losses in
-anonymity if clients still hold redemption tokens for previously active
-Issuers.
-
-Alternatively, when applicable, trusted registries can indicate which Issuers
-are deemed to be active. If a Client is asked to invoke the issuance protocol
-for an Issuer that is not declared active, then the client can refuse to run
-the protocol and obtain a token.
-
-Another option to allow a large number of Issuers in the ecosystem,
-while preventing the joining of a number of different tokens, is for the
-Client to maintain sharded "redemption partitions". Within each partition,
-the Client would place an upper bound on the number of allowed Issuers.
-As long as each redemption partition maintains a strong privacy boundary
-with the others, the number of bits of information the Origin can learn is
-bounded by the number of redemption partitions.
-
-## Side-Channel Attacks
+the same configuration. Issuer configuration updates, e.g., due to key rotation,
+are an important part of hedging against long-term private key compromise.
+In general, key rotations represent a trade-off between Client
+privacy and Issuer security. Therefore, it is important that key rotations occur
+on a regular cycle to reduce the harm of an Issuer key compromise.
+
+Lastly, if there are a large number of selected or trusted Issuers,
+and Origins accept all of them, segregation can occur. If Clients obtain tokens
+from many Issuers, and Origins later challenge a Client for a token from each
+Issuer, Origins can learn information about the Client. Each per-Issuer token
+that a Client holds essentially corresponds to a bit of information about the
+Client that Origin learns. Therefore, there is an exponential loss in anonymity
+relative to the number of Issuers.
+
+The fundamental problem here is that the number of possible issuance configurations,
+including the keys in use and the Issuer identities themselves, can partition the Client
+anonymity set. To mitigate this problem, Clients SHOULD bound the number of active
+issuance configurations. Moreover, Clients SHOULD employ some form of consistency
+mechanism to ensure that they receive the same configuration information and are
+not being actively partitioned into smaller anonymity sets. See {{CONSISTENCY}} for
```

Comment:
I guess my comment above could turn into another recommendation here to avoid using multiple issuers with a particular origin.